vcs/gitcmd/repo.go

```diff
@@ -962,6 +962,35 @@ func (r *Repository) Committers(opt vcs.CommittersOptions) ([]*vcs.Committer, er
 	return committers, nil
 }
 
+func (r *Repository) ListFiles(at vcs.CommitID) ([]string, error) {
+	if err := checkSpecArgSafety(string(at)); err != nil {
+		return nil, err
+	}
+
+	r.editLock.RLock()
+	defer r.editLock.RUnlock()
+
+	if at == "" {
+		at = "HEAD"
+	}
+
+	cmd := exec.Command("git", "ls-files", "--with-tree", string(at))
+	cmd.Dir = r.Dir
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("exec `git ls-files --with-tree %v` failed: %v", at, err)
+	}
+	out = bytes.TrimSpace(out)
```

Comment:
This feels a little broad and imprecise of a tool for the job. You're just using it to prevent the last element from being a blank line, since the output is newline terminated and that happens when you use `Split(out, "\n")`, right?

How about just doing `out = bytes.TrimSuffix(s, []byte("\n"))` instead?